JavadocTypeCheck incorrectly matches record component @param tags using prefix instead of exact match

I noticed that the Javadoc check does not correctly validate `@param` tags for record components.

Currently, it treats a tag as valid if it shares a prefix with the component name, instead of requiring an exact match. This leads to inconsistent validation results.

---

### **Steps to reproduce**

1. Create a file with the following content:

```java
/**
 * Example case
 *
 * @param john123 wrong tag for a different component
 */
record Sample(String john) {}
```

2. Use the following Checkstyle configuration:

```xml
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType"/>
    </module>
</module>
```

3. Run Checkstyle:

```bash
checkstyle -c checkstyle.xml Sample.java
```

---

### **Expected behavior**

- `@param` tags should match the record component name **exactly**
- A missing `@param john` violation should be reported

---

### **Actual behavior**

- `@param john123` is treated as documenting `john` due to prefix-based matching  
- Missing `@param john` is **not reported**  
- `john123` is still flagged as an unused tag  

---

### **Why this is a problem**

This creates inconsistent behavior:
- The incorrect tag is accepted in one place  
- But rejected in another  

This makes the validation confusing and unreliable.

---

Hi @romani 

I have read check documentation: https://checkstyle.org/checks/javadoc/javadocype.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

➜  Checkstyle javac Sample.java
➜  Checkstyle cat checkstyle.xml
```xml
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocType"/>
  </module>
</module>%
```
➜  Checkstyle cat Sample.java
```java
/**
 * Example case
 *
 * @param john123 wrong tag for a different component
 */
record Sample(String john) {}%
```
➜  Checkstyle RUN_LOCALE="-Duser.language=en -Duser.country=US"
➜  Checkstyle java $RUN_LOCALE -jar checkstyle-13.4.2-all.jar -c checkstyle.xml Sample.java
Starting audit...
[ERROR] /Users/mgusti.maulanaz/Documents/Checkstyle/Sample.java:4:4: Unused @param tag for 'john123'. [JavadocType]
Audit done.
Checkstyle ends with 1 errors.

---

When a record has a component named `john` and its Javadoc contains `@param john123` (a tag for a different component, not `john`), `JavadocTypeCheck` should NOT report a missing `@param john` tag. The `@param john123` tag should only be flagged as an unused tag for a different component.

`JavadocTypeCheck` incorrectly matches `@param john123` to the record component `john` using prefix matching (`indexOf("john") == 0`), causing it to:
1. Report a false positive "missing @param john tag" error
2. Fail to correctly identify `@param john123` as an unused tag for a different component

**Expected behavior:** `@param` tags should match record components by exact name (or name + space for additional description), not by prefix. For example, `@param john123` should NOT match record component `john` because they are different parameter names.

---

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java
@@ -407,8 +407,11 @@ public class JavadocTypeCheck
 
         final boolean found = tags
             .stream()
-            .filter(JavadocTag::isParamTag)
-            .anyMatch(tag -> tag.getFirstArg().indexOf(recordComponentName) == 0);
+                .filter(JavadocTag::isParamTag).anyMatch(tag -> {
+                    final String arg = tag.getFirstArg();
+                    return arg.equals(recordComponentName)
+                            || arg.startsWith(recordComponentName + SPACE);
+                });
 
         if (!found) {
             log(ast, MSG_MISSING_TAG, JavadocTagInfo.PARAM.getText()

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheckTest.java
@@ -370,6 +370,23 @@ public class JavadocTypeCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testJavadocTypeRecordComponentNameMismatch() throws Exception {
+        final String[] expected1 = {
+            "21:4: " + getCheckMessage(MSG_UNUSED_TAG, "@param", "valueExtra"),
+            "23:1: " + getCheckMessage(MSG_MISSING_TAG, "@param value"),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputJavadocTypeRecordComponentNameMismatch.java"), expected1);
+
+        final String[] expected2 = {
+            "23:1: " + getCheckMessage(MSG_MISSING_TAG, "@param value"),
+            "21:4: " + getCheckMessage(MSG_UNUSED_TAG_GENERAL),
+        };
+        verifyWithInlineConfigParser(
+                getPath("InputJavadocTypeRecordComponentNameMismatch2.java"), expected2);
+    }
+
+    @Test
     public void testJavadocTypeParamDescriptionWithAngularTags() throws Exception {
         final String[] expected = {
             "50:4: " + getCheckMessage(MSG_UNUSED_TAG, "@param", "<P>"),

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctype/InputJavadocTypeRecordComponentNameMismatch.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctype/InputJavadocTypeRecordComponentNameMismatch.java
@@ -1,0 +1,25 @@
+/*
+JavadocType
+scope = protected
+excludeScope = (default)null
+authorFormat = (default)null
+versionFormat = (default)null
+allowMissingParamTags = (default)false
+allowUnknownTags = (default)false
+allowedAnnotations = (default)Generated
+tokens = (default)INTERFACE_DEF, CLASS_DEF, ENUM_DEF, ANNOTATION_DEF, RECORD_DEF
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadoctype;
+
+// violation 4 lines below 'Unused @param tag for 'valueExtra'.'
+/**
+ * Example case
+ *
+ * @param valueExtra wrong tag for a different component
+ */
+public record InputJavadocTypeRecordComponentNameMismatch(String value) {
+    // violation above 'Type Javadoc comment is missing @param value tag.'
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctype/InputJavadocTypeRecordComponentNameMismatch2.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadoctype/InputJavadocTypeRecordComponentNameMismatch2.java
@@ -1,0 +1,25 @@
+/*
+JavadocType
+scope = protected
+excludeScope = (default)null
+authorFormat = (default)null
+versionFormat = (default)null
+allowMissingParamTags = (default)false
+allowUnknownTags = (default)false
+allowedAnnotations = (default)Generated
+tokens = (default)INTERFACE_DEF, CLASS_DEF, ENUM_DEF, ANNOTATION_DEF, RECORD_DEF
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadoctype;
+
+// violation 4 lines below 'Unused Javadoc tag.'
+/**
+ * Example case with bare @param tag
+ *
+ * @param
+ */
+public record InputJavadocTypeRecordComponentNameMismatch2(String value) {
+    // violation above 'Type Javadoc comment is missing @param value tag.'
+}

